src/main/java/com/altenheim/kalender/controller/logicController/ImportController.java

```diff
@@ -14,5 +44,111 @@ public ImportController(ICalendarEntriesModel allEntries, IWebsiteScraperControl
+            }
+            var start = (DtStart)(components.get(i).getProperties().getProperty("DTSTART"));
+            var end = (DtEnd)((Property) components.get(i).getProperties().getProperty("DTEND"));
+            var startMilli = start.getDate().toInstant().toEpochMilli();
+            var endMilli = end.getDate().toInstant().toEpochMilli();            
+            var entry = createCalendarFXEntryFromMillis(startMilli, endMilli);
+            var summary = ((Property) components.get(i).getProperties().getProperty("SUMMARY")).getValue();
+            entry.setTitle(summary);
+            var locationProp = (Property) components.get(i).getProperties().getProperty("LOCATION");
+            if(locationProp != null)
+                entry.setLocation(locationProp.getValue());
+            cal.addEntry(entry);
+        }
+        System.out.println(cal.findEntries("").get(0).getInterval().toString());
+        return cal;
+    } 
+
+    public void exportFile(com.calendarfx.model.Calendar cal) throws ValidationException, IOException
+    {
+        var entries = cal.findEntries("");
+        var calName = cal.getName();
+        var icsCalendar = new Calendar();
+        icsCalendar.getProperties().add(new ProdId("-//Smart Planner//iCal4j 1.0//DE"));
+        icsCalendar.getProperties().add(Version.VERSION_2_0);
+        icsCalendar.getProperties().add(CalScale.GREGORIAN);
+        icsCalendar.getProperties().add(new XProperty("X-WR-CALNAME", calName));
+        for(int i = 0; i < entries.size(); i++)
+        {
+            var entry = entries.get(i);
+            icsCalendar.getComponents().add(createIcalEntryFromCalFXEntry(entry));
+        }
+        var fout = new FileOutputStream("testExport.ics");
+        var outputter = new CalendarOutputter();
+        outputter.output(icsCalendar, fout);
+        System.out.println("File created in root directory.");
+    }
+
+    private Entry<String> createCalendarFXEntryFromMillis(long start, long end)
+	{
+		var entry = new Entry<String>();
+		var dateStart = LocalDateTime.ofInstant(Instant.ofEpochMilli(start), ZoneId.systemDefault());
+		var dateEnd = LocalDateTime.ofInstant(Instant.ofEpochMilli(end), ZoneId.systemDefault());		
+		entry.changeStartTime(dateStart.toLocalTime());
+		entry.changeStartDate(dateStart.toLocalDate());
+		entry.changeEndTime(dateEnd.toLocalTime());
+		entry.changeEndDate(dateEnd.toLocalDate());
+		return entry;
+	} 
     
+    private VEvent createIcalEntryFromCalFXEntry(Entry<?> entry)
+    {
+        var startTime = GregorianCalendar.from(entry.getStartAsZonedDateTime()).getTime();
+        var endTime = GregorianCalendar.from(entry.getEndAsZonedDateTime()).getTime();
+        var start = new DateTime(startTime);
+        var end = new DateTime(endTime);
+        var title = entry.getTitle();
+        var event = new VEvent(start, end, title);
+        var iD = new RandomUidGenerator();
+        var uid = iD.generateUid();
+        event.getProperties().add(uid); 
```

Comment:
Edgecase: zufällig wurde die ID vorher schon generiert, wird der von der Methode abgedeckt=

Reply:
geht nicht, dafür sorgt der generator. also kein edge case sondern out of scope